### PR TITLE
FK templates

### PR DESCRIPTION
**Pull request by waveform80**
Ensure that any flash-kernel templates that appear in the boot-scripts are substituted correctly (mostly blank, but there's a couple of exceptions). The only tricky one is the @@KERNEL_VERSION@@ template (which we don't actually use even in the new boot-script, but we might as well implement in case it ever gets used in the future). In this case it gets taken from the name of a dir in the linux-modules-*-raspi2 package - which is a bit of a hack but should be reliable enough?

Also tidied up device-tree and overlay copying (arm64 and armhf bizarrely have different dirs for their device-trees and the former code wound up making a redundant copy of them in the arm64 case), and removed the dtoverlay which was preventing the 3A+ from booting (this change requires the new kernel intended to fix the Pi4 USB issue).

### CODE REVIEW

**Review by sil2100 — APPROVED** [review]
Looks good, and thanks for cleaning up the .dtb copy bits. At first I was worried about implications of removing the dtoverlay without the new kernel, but the impact seems to be only on focal - which is anyway not guaranteed to be 'working'.

**Comment by vorlonofportland** [comment]
> Looks good, and thanks for cleaning up the .dtb copy bits. At first I was worried about implications of removing the dtoverlay without the new kernel, but the impact seems to be only on focal - which is anyway not guaranteed to be 'working'.

Given that this is for classic, I would certainly hope we are keeping the focal images working and not regressing them.

**Comment by waveform80** [comment]
> Given that this is for classic, I would certainly hope we are keeping the focal images working and not regressing them.

The effect of this on Focal should be as follows:

1. The Linux framebuffer will break (de-sync) until the kernel fix from [LP: #1848790](https://bugs.launchpad.net/ubuntu/+source/linux-raspi2/+bug/1848790) is merged
2. The u-boot framebuffer will work again on all pis (currently it's broken, due to the fkms dtoverlay)
3. The 3A+ will be able to boot (the fkms dtoverlay is incompatible with the 3A+'s reduced memory)

In other words, it'll fix some things and (temporarily) break one.

**Comment by sil2100** [comment]
Yes, no worries Steve. By 'not guaranteed to be working' I meant that we can allow having our dailies not operational for a day or two. It is most certainly required for us to have it working in the long run though.
